run/test/specs/utils/create_contact.ts

```diff
@@ -13,57 +14,54 @@ export const newContact = async (
   await device1.sendNewMessage(Bob, `${Alice.userName} to ${Bob.userName}`);
   // Click on message request folder
   await sleepFor(100);
-  await retryRequest('Message requests banner', platform, device1, device2);
+  await runOnlyOnIOS(platform, () => retryMsgSentForBanner(platform, device1, device2, 30000));
   await device2.clickOnByAccessibilityID('Message requests banner');
-  // Select message from User A
-  await retryRequest('Message request', platform, device1, device2);
   await device2.clickOnByAccessibilityID('Message request');
   await runOnlyOnAndroid(platform, () =>
     device2.clickOnByAccessibilityID('Accept message request')
   );
   // Type into message input box
   await device2.sendMessage(`Reply-message-${Bob.userName}-to-${Alice.userName}`);
   // Verify config message states message request was accepted
+  // "messageRequestsAccepted": "Your message request has been accepted.",
+  const messageRequestsAccepted = englishStripped('messageRequestsAccepted').toString();
+  await device1.waitForControlMessageToBePresent(messageRequestsAccepted as ControlMessage);
   await device1.waitForControlMessageToBePresent('Your message request has been accepted.');
 
   console.info(`${Alice.userName} and ${Bob.userName} are now contacts`);
   return { Alice, Bob, device1, device2 };
 };
 
-export const retryRequest = async (
-  requestType: 'Message requests banner' | 'Message request',
+const retryMsgSentForBanner = async (
   platform: SupportedPlatformsType,
   device1: DeviceWrapper,
   device2: DeviceWrapper,
-  timeout: number = 10000
+  timeout: number
 ) => {
   const startTime = Date.now();
   let messageRequest: boolean | null = false;
-  if (platform === 'ios') {
-    while (!messageRequest && Date.now() - startTime < timeout) {
-      const element = await device2.doesElementExist({
-        strategy: 'accessibility id',
-        selector: requestType,
-        maxWait: 1000,
-      });
 
-      messageRequest = element !== null;
+  while (!messageRequest && Date.now() - startTime < timeout) {
+    const element = await device2.doesElementExist({
+      strategy: 'accessibility id',
+      selector: 'Message requests banner',
+      maxWait: 5000,
+    });
 
-      if (!messageRequest) {
-        await device1.sendMessage('Retry');
-        console.log(`Retrying message request`);
-        await sleepFor(1000);
-      } else {
-        console.log('Found message request: No need for retry');
-      }
-    }
+    messageRequest = element !== null;
 
     if (!messageRequest) {
-      throw new Error(
-        'Message request did not appear within the timeout period: This is a common race condition on iOS.'
-      );
+      await device1.sendMessage('Retry');
+      console.log(`Retrying message request`);
```

Comment:
the console.log should probably be before the sendMessage.
Just because if the send takes 10s, you don't know what is happening from the logs until after it passed/failed.

Reply:
Fixed